srml/contract/src/exec.rs

```diff
@@ -400,12 +415,29 @@ where
 					EmptyOutputBuf::new(),
 					gas_meter,
 				)
-				.into_result()?;
+				.into_result();
+
+			// Log an exeuction failure event
+			if let Err(err) = result {
+				self.events.push(RawEvent::ExecutionFailed(
```

Comment:
It is possible to create a contract (malicious or not) that just fails which leads to adding an event. It might be possible to spend all block gas budget to spam the event list. I'm not sure if it would be possible to assign a reasonable price for it without penalizing the normal path.

I'm afraid that I can't come up with a solution for this one at the moment. May be we should take a totally different route to solve this issue. For example, we might want to run a smart-contract with small or no amount of diagnostic tools enabled, since it seems to me that it is overhead to build consensus on that. Instead, we could add a special running mode that would allow a user to replay the failing transaction locally recording all possible diagnostics.